samples/junit-tests/src/test/java/com/yelp/codegen/generatecodesamples/PostFileEndpointTest.kt

```diff
@@ -0,0 +1,179 @@
+package com.yelp.codegen.generatecodesamples
+
+import com.yelp.codegen.generatecodesamples.apis.FileApi
+import com.yelp.codegen.generatecodesamples.tools.MockServerApiRule
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+private data class FileApiStats(
+    var boundary: String? = null,
+    var errors: List<String>? = null,
+    var parts: List<MultiPartInfo>? = null
+)
+private data class MultiPartInfo(
+    val content: String? = null,
+    val contentDisposition: String? = null,
+    val contentLength: String? = null,
+    val contentType: String? = null
+)
+
+private fun processPart(part: String): MultiPartInfo {
+
+    val partLines = part.split("\r\n").let {
+        // part would look like "\r\nHeader-Name: Header-Value\r\n...\r\n\r\nbody\r\n"
+        // Here we're removing the first and last part as we know that are empty and
+        it.subList(1, it.size - 1)
+    }
+
+    val partLinesIterator = partLines.iterator()
+    val headers = mutableMapOf<String, String>()
+    for (headerLine in partLinesIterator) {
+        if (headerLine.isEmpty()) {
+            break
+        } else {
+            val (name, value) = headerLine.split(": ", limit = 2)
+            headers[name] = value
+        }
+    }
+
+    val content = partLinesIterator.asSequence().joinToString("\r\n")
+    return MultiPartInfo(
+        content = content,
+        contentDisposition = headers["Content-Disposition"],
+        contentType = headers["Content-Type"],
+        contentLength = headers["Content-Length"]
+    )
+}
+
+private val RecordedRequest.fileApiStats: FileApiStats
+    get() {
+        /**
+         * requestBody has to be compliant to the HTTP Specifications for MultiPart (RFC 2387)
+         *
+         * This means that the body will look like
+         *      --<BOUNDARY>\r\n
+         *      <Content1 Header Name>: <Content1 Header Value>\r\n
+         *      [<Content1 Header Name>: <Content1 Header Value>\r\n]
+         *      \r\n
+         *      <Content1>\r\n
+         *      --<BOUNDARY>\r\n
+         *      <Content# Header Name>: <Content# Header Value>\r\n
+         *      [<Content# Header Name>: <Content# Header Value>\r\n]
+         *      \r\n
+         *      <Content#>\r\n
+         *      --<BOUNDARY>--\r\n
+         *      \r\n
+         */
+        val requestBody = this.body.readUtf8()
+
+        val boundaryLine = requestBody.splitToSequence("\r\n").first()
```

Comment:
```suggestion
        val boundaryLine = requestBody.split("\r\n").first()
```

Reply:
I was thinking to use sequences in order to avoid the creation of the `String` related to the second part of the split as we don't use it and it is basically all the body.
I can agree that we're writing a test, that is going to ingest a limited amount of data, so it should be ok ™️